modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheProxyImpl.java

```diff
@@ -566,8 +569,10 @@ private ClusterGroup projection(boolean loc) {
         else {
             ContinuousQueryWithTransformer<K, V, ?> qry0 = (ContinuousQueryWithTransformer<K, V, ?>)qry;
 
-            if (qry0.getLocalListener() == null)
-                throw new IgniteException("Mandatory local transformed event listener is not set for the query: " + qry);
+            if (qry0.getLocalListener() == null && qry0.getRemoteFilterFactory() == null) {
```

Comment:
May be check on qry0.getRemoteFilter() is also necessary here?

Reply:
ContinuousQueryWithTransformer doesn't provide this method.